File: backend/app/models/guardian.py
from datetime import datetime
from typing import TYPE_CHECKING
from uuid import UUID, uuid4

from sqlmodel import Field, Relationship, SQLModel

if TYPE_CHECKING:
    from .child import Child

class Guardian(SQLModel, table=True):
    __tablename__ = "guardians"

    id: UUID = Field(default_factory=uuid4, primary_key=True)
    email: str = Field(unique=True, index=True)
    hashed_password: str
    first_name: str
    last_name: str
    preferred_language: str = Field(default="ar")
    timezone: str = Field(default="Asia/Dubai")
    is_email_verified: bool = Field(default=False)
    is_active: bool = Field(default=True)

    # Timestamps
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime | None = Field(default=None)
    last_login: datetime | None = Field(default=None)

    # Relationships
    children: list["Child"] = Relationship(back_populates="guardian")

    class Config:
        validate_assignment = True

File: packages/schemas/src/guardian.ts
import { z } from 'zod'

// Guardian schema for AI Education Platform
export const GuardianSchema = z.object({
  id: z.string().uuid(),
  email: z.string().email('Invalid email format'),
  firstName: z.string().min(1, 'First name is required'),
  lastName: z.string().min(1, 'Last name is required'),
  preferredLanguage: z.enum(['ar', 'en']).default('ar'),
  timezone: z.string().default('Asia/Dubai'),
  isEmailVerified: z.boolean().default(false),
  isActive: z.boolean().default(true),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime().optional(),
  lastLogin: z.string().datetime().optional(),
})

export const CreateGuardianSchema = GuardianSchema.omit({
  id: true,
  isEmailVerified: true,
  isActive: true,
  createdAt: true,
  updatedAt: true,
  lastLogin: true,
})

export const UpdateGuardianSchema = CreateGuardianSchema.partial()

export type Guardian = z.infer<typeof GuardianSchema>
export type CreateGuardian = z.infer<typeof CreateGuardianSchema>
export type UpdateGuardian = z.infer<typeof UpdateGuardianSchema>
